packages/babel-preset/index.js

```diff
@@ -114,5 +109,10 @@ module.exports = function shopifyCommonPreset(
       require.resolve('@babel/plugin-transform-react-constant-elements'),
   ].filter(Boolean);
 
-  return {presets, plugins};
+  const assumptions = {
+    setPublicClassFields: true,
```

Comment:
I think these should only be set when `typescript` is true, as before we only enabled loose mode when typescript was true